lib/flat_search/olx_scraper.ex

```diff
@@ -0,0 +1,116 @@
+defmodule OlxScraper do
+  use Crawly.Spider
+  alias FlatSearch.Flats
+
+  @impl Crawly.Spider
+  def base_url(), do: "https://www.olx.pl"
+
+  @impl Crawly.Spider
+  def init(), do: [start_urls: ["https://www.olx.pl/nieruchomosci/mieszkania/wynajem/"]]
+
+  @impl Crawly.Spider
+  def parse_item(res) do
+    # First page only
+    {:ok, document} = Floki.parse_document(res.body)
+
+    flat_urls =
+      document
+      |> Floki.find("tr td div table tbody tr td a.link")
+      |> Floki.attribute("href")
+      |> Enum.reject(&(!String.contains?(&1, base_url())))
```

Comment:
maybe instead of `reject` and `!` `Enum.filter` would be sufficient, so you don't need to reject based of negated value, just filter based of value